pkg/kubelet/kubelet.go

```diff
@@ -2483,6 +2489,13 @@ func (kl *Kubelet) canAdmitPod(pods []*api.Pod, pod *api.Pod) (bool, string, str
 			glog.V(2).Infof("Predicate failed on Pod: %v, for reason: %v", format.Pod(pod), message)
 			return fit, reason, message
 		}
+		if re, ok := err.(*predicates.ErrTaintsTolerationsNotMatch); ok {
```

Comment:
I'm not sure about this check. Correct me if I'm wrong but for now GeneralPredicates does not check Taints. There is a separate function for it `predicates.PodToleratesNodeTaints` so this error will not appear here.